src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -368,22 +357,34 @@ export const useBalanceChartData: UseBalanceChartData = args => {
   const selectedCurrency = useAppSelector(selectSelectedCurrency)
 
   // kick off requests for all the price histories we need
-  useFetchPriceHistories({ assetIds, timeframe })
+  useFetchPriceHistories(assetIds, timeframe)
   const cryptoPriceHistoryData = useAppSelector(state =>
     selectCryptoPriceHistoryTimeframe(state, timeframe),
   )
 
   const fiatPriceHistoryData = useAppSelector(state =>
     selectFiatPriceHistoryTimeframe(state, timeframe),
   )
-  // loading state
-  useEffect(() => setBalanceChartDataLoading(true), [setBalanceChartDataLoading, timeframe])
+
+  const isTxHistoryAvailableFilter = useMemo(
+    () => ({ accountId, timeframe }),
+    [accountId, timeframe],
+  )
+  const isTxHistoryAvailable = useAppSelector(state =>
+    selectIsTxHistoryAvailableByFilter(state, isTxHistoryAvailableFilter),
+  )
 
   // calculation
-  useEffect(() => {
+  return useMemo(() => {
```

Comment:
This feels weird - if everything is nice and memoized as it should, the logic here should be part of the "render function" (or whatever the return branch is called for hooks)

Reply:
The idea here is to memoize the object being returned. If we do the following
```
const { fieldA, fieldB } = useMemo(() => { ...

return { fieldA, fieldB }
```
...we are effectively negating the utility of memoization because it's creating a new object every time a prop changes, which in turn triggers a rerender.

We have 2 options:
1. return the `useMemo` like as is
2. disregard and just let it rip 😅

Reply:
Perhaps the comment wasn't clear, my bad! 

The issue here is that `useMemo()` should be used to memoize an expression, not being returned by itself. At the end of the day, it *is* an effect hook and should never be returned directly.
The whole purpose of the render method should be to return some value (not a `useEffect`, `useMemo` and other effect hooks call, though a var that's assigned to a `useMemo` call is absolutely fine), which is why this feels weird.

- If memoizing the *whole* bit was the intent, then you could do `const value = useMemo()` and return the value, which is a simple change but will keep things the react way.

Happy if you want to do this as a two-liner diff (minus the whitespaces!), though looking at the diff it still feels off. Render logic (or "return" logic in this case for hooks) should happen at render method time:

We effectively have 2 branches here.

A first early return here:

https://github.com/shapeshift/web/blob/f545d8d7368f4f950738504e29a803ff90e306f1/src/hooks/useBalanceChartData/useBalanceChartData.ts#L379-L388

Another there:

https://github.com/shapeshift/web/blob/f545d8d7368f4f950738504e29a803ff90e306f1/src/hooks/useBalanceChartData/useBalanceChartData.ts#L390-L412

Assuming the difference vars (`emptyBuckets` and `const balanceChartData = bucketsToChartData(calculatedBuckets)` namely) are nice and memoized with `useMemo()` and don't happen at render method time directly, then the whole `if (cond) {return left}; // doLogic return right` bit could and should happen at render method time i.e at the end of the hook after all hooks (`useMemo`, `useEffect`) run.

re: The idea here is to memoize the object being returned. If we do the following

We can absolutely do that by memoizing both the left and right side to ensure referencial stablity i.e `loadingBalanceChartData` for the left loading side, and `loadedBalanceChartData` for the right, loaded side.